### PR TITLE
fix: interpret proto cross-reference syntax

### PR DESCRIPTION
**Pull request by dazuma**
AIP 192 describes [a format for cross-references in comments](https://aip.dev/192#cross-references). This PR detects cross-references and replaces them with the corresponding YARD syntax for generating links to the appropriate documentation.

Note: we treat `google.longrunning.Operations` as a special case, because the generator creates special operations clients that live in the generated service namespace rather than under `Google::Longrunning::Operations`. Because there might be multiple such operations clients, and it might be ambiguous which one to use, we punt and just provide the text without a yard reference.

### CODE REVIEW

File: gapic-generator/lib/gapic/formatting_utils.rb [hunk 1]
```diff
@@ -104,6 +103,42 @@ def escape_line_braces line
...
+          return line if file.nil?
+          transformed_xref = transform_xref api, entity, file, m[:text]
+          return line if transformed_xref.nil?
+          line = "#{m[:pre]}#{transformed_xref}#{m[:post]}"
+        end
+        line
+      end
+
+      def transform_xref api, entity, file, text
+        return "`#{text}`" if entity.address[0, 3] == ["google", "longrunning", "Operations"]
```
Comment (viacheslav-rostovtsev) [inline gapic-generator/lib/gapic/formatting_utils.rb]: would appreciate a comment on why we are carving a specific execution path for these tokens

File: gapic-generator/lib/gapic/schema/wrappers.rb [hunk 2]
```diff
@@ -376,6 +388,10 @@ def initialize descriptor, address, docs, messages, enums, services,
         @services.each { |m| m.parent = self }
       end
 
+      def containing_file
+        self
+      end
```
Comment (viacheslav-rostovtsev) [inline gapic-generator/lib/gapic/schema/wrappers.rb]: duplication?

File: gapic-generator/test/gapic/formatting_utils_test.rb [hunk 3]
```diff
@@ -239,26 +239,112 @@ def test_escape_braces_separators_capitalized
...
+      api.add_file! "google.cloud.example" do
+        api.add_service! "Earth"
+      end
+    end
+    result = Gapic::FormattingUtils.format_doc_lines api, ["Hello, [World][google.cloud.example.Earth]!\n"]
+    assert_equal ["Hello, {Google::Cloud::Example::Earth::Client World}!\n"], result
+  end
+
+  def test_xref_with_method
+    api = FakeApi.new do |api|
```
Comment (viacheslav-rostovtsev) [inline gapic-generator/test/gapic/formatting_utils_test.rb]: Missing a test for the Field out of the Service/Message/Method/Field switch block in transform_xref

File: gapic-generator/test/test_helper.rb [hunk 4]
```diff
@@ -150,3 +151,113 @@ def service api_name, service_name
     service_obj
   end
 end
+
+# A fake api class for creating test fixtures
+class FakeApi
```
Comment (viacheslav-rostovtsev) [inline gapic-generator/test/test_helper.rb]: :+1:

File: shared/output/cloud/speech_v1/lib/google/cloud/speech/v1/speech/operations.rb [hunk 5]
```diff
@@ -324,11 +324,11 @@ def delete_operation request, options = nil
             #     makes a best effort to cancel the operation, but success is not
             #     guaranteed.  If the server doesn't support this method, it returns
             #     `google.rpc.Code.UNIMPLEMENTED`.  Clients can use
-            #     [Operations.GetOperation][google.longrunning.Operations.GetOperation] or
+            #     `Operations.GetOperation` or
             #     other methods to check whether the cancellation succeeded or whether the
             #     operation completed despite cancellation. On successful cancellation,
             #     the operation is not deleted; instead, it becomes an operation with
-            #     an [Operation.error][google.longrunning.Operation.error] value with a [google.rpc.Status.code][google.rpc.Status.code] of 1,
+            #     an {Google::Longrunning::Operation#error Operation.error} value with a {Google::Rpc::Status#code google.rpc.Status.code} of 1,
```
Comment (viacheslav-rostovtsev) [inline shared/output/cloud/speech_v1/lib/google/cloud/speech/v1/speech/operations.rb]: Wasn't the intent in the code to remove the Google::Longrunning?

File: gapic-generator/lib/gapic/schema/wrappers.rb (hunk 2 above)
Comment (dazuma) [reply gapic-generator/lib/gapic/schema/wrappers.rb]: No, this is intentional. The goal is to return the `Gapic::Schema::File` object in the tree. In the base class, we turn around and ask the parent, but if the node is a File, we override that to return self.

File: shared/output/cloud/speech_v1/lib/google/cloud/speech/v1/speech/operations.rb (hunk 5 above)
Comment (dazuma) [reply shared/output/cloud/speech_v1/lib/google/cloud/speech/v1/speech/operations.rb]: The intent is to omit specifically the _service_ `Google::Longrunning::Operations` (plural), but not anything else under `Google::Longrunning` including in this case, the _message_ `Google::Longrunning::Operation` (singular).

Comment (viacheslav-rostovtsev) [reply shared/output/cloud/speech_v1/lib/google/cloud/speech/v1/speech/operations.rb]: well, this really needs to be a comment in `transform_xref`

File: gapic-generator/lib/gapic/formatting_utils.rb (hunk 1 above)
Comment (dazuma) [reply gapic-generator/lib/gapic/formatting_utils.rb]: done

File: gapic-generator/test/gapic/formatting_utils_test.rb (hunk 3 above)
Comment (dazuma) [reply gapic-generator/test/gapic/formatting_utils_test.rb]: good catch. done.

**Pull request merged**